### PR TITLE
Add check to ensure job/group uuids do not exist before creation

### PR DESCRIPTION
**Pull request by pschorf**

### CODE REVIEW

**Comment by wyegelwel** [comment]
What error does this return if we catch the conflict at transaction time?

**Comment by pschorf** [comment]
This throws an ExecutionException wrapping an IllegalStateException. That will cause liberator to return a 503

**Comment by wyegelwel** [comment]
> This throws an ExecutionException wrapping an IllegalStateException. That will cause liberator to return a 503

I imagine the answer is no, liberator is too rigid, but is there a way to get it to return a 409 conflict like it would in the case that cook catches the conflict up front?

File: scheduler/src/cook/mesos/schema.clj [hunk 1]
```diff
@@ -945,7 +945,17 @@ for a job. E.g. {:resources {:cpus 4 :mem 3} :constraints {\"unique_host_constra
                          prior-state (:job/state job)]
                      (if (= prior-state :job.state/completed)
                        [[:db/add j :job/state :job.state/waiting]]
-                       []))}}])
+                       []))}}
+   {:db/id (d/tempid :db.part/user)
+    :db/ident :ensure-not-exists
+    :db/doc "Ensure that the given entity does not exist"
+    :db/fn #db/fn {:lang "clojure"
+                   :params [db field id]
```
Comment (wyegelwel) [inline scheduler/src/cook/mesos/schema.clj]: Change this to take `db` and `id-or-lookup-ref` and then call `(d/entity db id-or-lookup-ref)` that way it is more generic.

File: scheduler/src/cook/mesos/schema.clj [hunk 2]
```diff
@@ -945,7 +945,17 @@ for a job. E.g. {:resources {:cpus 4 :mem 3} :constraints {\"unique_host_constra
                          prior-state (:job/state job)]
                      (if (= prior-state :job.state/completed)
                        [[:db/add j :job/state :job.state/waiting]]
-                       []))}}])
+                       []))}}
+   {:db/id (d/tempid :db.part/user)
+    :db/ident :ensure-not-exists
```
Comment (wyegelwel) [inline scheduler/src/cook/mesos/schema.clj]: It is datomic best practices to namespace idents. Perhaps `:entity/ensure-not-exists`

File: scheduler/src/cook/mesos/schema.clj (hunk 1 above)
Comment (pschorf) [inline scheduler/src/cook/mesos/schema.clj]: Done

File: scheduler/src/cook/mesos/schema.clj (hunk 2 above)
Comment (pschorf) [inline scheduler/src/cook/mesos/schema.clj]: Done

**Comment by pschorf** [comment]
I don't think there's a way to affect the response code from liberator. Once it decides to create the entity, the response code is always 201 unless you throw in which case it is 500 (not 503.)

**Pull request merged**